src/pages/Main/RegionButton/RegionButton.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import Modal from './RegionModal/RegionModal';
+import styled from 'styled-components';
+
+function Button({ title, description, tagFetchHandler, tagFetchData }) {
+  const [modalOn, setModalOn] = useState(false);
+
+  const handleModal = () => {
+    setModalOn(!modalOn);
+  };
```

Comment:
```suggestion
  const handleModal = () => {
    setModalOn(prevModalOn => !prevModalOn);
  };
```
이전 스테이트 기준으로 변해야하는 값이라면 인자로 함수를 전달하면,
전달한 함수의 인자로 prevState를 주입해줍니다!
이런식으로 인자로 전달받은 prevState는 이 setModalOn 함수가 호출될 때의 최신의 스테이트값이라고 확신할 수 있기에 좀 더 안정적입니다! 참고해보세요!

Reply:
넵! 감사합니다!